pdfbox/src/main/java/org/apache/pdfbox/pdmodel/font/TrueTypeEmbedder.java

```diff
@@ -136,13 +136,17 @@ public final void buildFontFile2(InputStream ttfStream) throws IOException
     /**
      * Returns true if the fsType in the OS/2 table permits embedding.
      */
-    private boolean isEmbeddingPermitted(TrueTypeFont ttf) throws IOException
+    protected boolean isEmbeddingPermitted(TrueTypeFont ttf) throws IOException
     {
         if (ttf.getOS2Windows() != null)
         {
             int fsType = ttf.getOS2Windows().getFsType();
-            if ((fsType & OS2WindowsMetricsTable.FSTYPE_RESTRICTED) ==
-                             OS2WindowsMetricsTable.FSTYPE_RESTRICTED)
+            // use two's complement arithmetic to check if fsType is a power of 2
+            boolean powerOfTwo = fsType != 0 && (fsType & (fsType - 1)) == 0;
+            if (
+                    powerOfTwo &&
```

Comment:
I would be inclined to first grab the table version and then perhaps have a method for each. Some of that code may be shared since they are similar. I think you need to first identify the version before you can accurately apply the rules set forth in the MS article.

Reply:
Alright. As long as all the extant versions are accounted for then I am fine. When it comes to specification/compliance code like this you always want to play it by the numbers.

Reply:
Added a comment above regarding the math you are using.

Reply:
Fixed

Reply:
@larrylynn-wf - I'm just throwing out an option if the maintainers are not keen on this approach.

It looks like there is already a constant defined for the bit holding print and preview permissions `OS2WindowsMetricsTable.FSTYPE_PREVIEW_AND_PRINT`. Maybe we could use that and try to follow a similar pattern to the current code with something like this?
```
if ((fsType & OS2WindowsMetricsTable.FSTYPE_PREVIEW_AND_PRINT) !=
                             OS2WindowsMetricsTable.FSTYPE_PREVIEW_AND_PRINT &&
                (fsType & OS2WindowsMetricsTable.FSTYPE_RESTRICTED) ==
                             OS2WindowsMetricsTable.FSTYPE_RESTRICTED)
```
My bit math is pretty rusty, but I think it will at least work for the case you outlined above where fstype is 6 (0110)

Reply:
I guess we'd also want `(fsType & OS2WindowsMetricsTable.FSTYPE_EDITIBLE) != OS2WindowsMetricsTable.FSTYPE_EDITIBLE`